packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -197,11 +432,69 @@ export class Table extends Resource {
       actions: ['dynamodb:ListStreams'],
       resourceArns: ['*'],
     });
- }
+  }
+
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs A `TableAttributes` object.
+   */
+  public static fromTableAttributes(scope: Construct, id: string, attrs: TableAttributes): ITable {
+
+    class Import extends TableBase {
+      public readonly tableName = this.parseTableName(scope, attrs);
+      public readonly tableArn = this.parseTableArn(scope, attrs);
+      public readonly tableStreamArn = attrs.tableStreamArn;
+
+      private parseTableName(construct: IConstruct, props: TableAttributes): string {
```

Comment:
These private methods are not needed. This logic should be part of the body of the `fromTableAttributes` function.